internal/providers/vsphere/pke/driver/cluster_creator.go

```diff
@@ -0,0 +1,420 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"emperror.dev/errors"
+	"go.uber.org/cadence/client"
+	corev1 "k8s.io/api/core/v1"
+
+	intPKE "github.com/banzaicloud/pipeline/internal/pke"
+	"github.com/banzaicloud/pipeline/internal/providers/vsphere/pke"
+	vspherePKE "github.com/banzaicloud/pipeline/internal/providers/vsphere/pke"
+	"github.com/banzaicloud/pipeline/internal/providers/vsphere/pke/driver/commoncluster"
+	"github.com/banzaicloud/pipeline/internal/providers/vsphere/pke/workflow"
+	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
+	pkgPKE "github.com/banzaicloud/pipeline/pkg/cluster/pke"
+	"github.com/banzaicloud/pipeline/src/auth"
+	"github.com/banzaicloud/pipeline/src/cluster"
+	"github.com/banzaicloud/pipeline/src/secret"
+)
+
+const pkeVersion = "0.4.14"
+const MasterNodeTaint = pkgPKE.TaintKeyMaster + ":" + string(corev1.TaintEffectNoSchedule)
+
+func MakeVspherePKEClusterCreator(
+	logger Logger,
+	config ClusterCreatorConfig,
+	k8sPreparer intPKE.KubernetesPreparer,
+	organizations OrganizationStore,
+	secrets ClusterCreatorSecretStore,
+	store pke.ClusterStore,
+	workflowClient client.Client,
+) VspherePKEClusterCreator {
+	return VspherePKEClusterCreator{
+		logger:           logger,
+		config:           config,
+		creationPreparer: MakeVspherePKEClusterCreationParamsPreparer(logger, k8sPreparer),
+		organizations:    organizations,
+		secrets:          secrets,
+		store:            store,
+		workflowClient:   workflowClient,
+	}
+}
+
+// VspherePKEClusterCreator creates new PKE-on-Vsphere clusters
+type VspherePKEClusterCreator struct {
+	logger           Logger
+	config           ClusterCreatorConfig
+	creationPreparer VspherePKEClusterCreationParamsPreparer
+	organizations    OrganizationStore
+	secrets          ClusterCreatorSecretStore
+	store            vspherePKE.ClusterStore
+	workflowClient   client.Client
+}
+
+type OrganizationStore interface {
+	Get(ctx context.Context, id uint) (auth.Organization, error)
+}
+
+type ClusterCreatorSecretStore interface {
+	secretStore
+
+	GetByName(organizationID uint, secretName string) (*secret.SecretItemResponse, error)
+}
+
+type ClusterCreatorConfig struct {
+	OIDCIssuerURL               string
+	PipelineExternalURL         string
+	PipelineExternalURLInsecure bool
+}
+
+type NodePool struct {
+	CreatedBy     uint
+	Name          string
+	Roles         []string
+	Labels        map[string]string
+	Size          int
+	AdminUsername string
+	VCPU          int
+	RamMB         int
```

Comment:
`Ram`, but it should be `RAM`: https://go.googlesource.com/lint/+/d0100b6bd8b389f0385611eb39152c4d7c3a7905/lint.go#788